src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        /// </summary>
+        public Vector3 AccelerometerBias => _accelerometerBias;
+
+        /// <summary>
+        /// Get or set the accelerometer range
+        /// </summary>
+        public AccelerometerRange AccelerometerRange
+        {
+            get => _accelerometerRange;
+
+            set
+            {
+                WriteRegister(Register.ACCEL_CONFIG, (byte)((byte)value << 3));
+                // We will cache the range to avoid i2c access everytime this data is requested
+                // This allow as well to make sure the stored data is the same as the one read
+                _accelerometerRange = (AccelerometerRange)(ReadByte(Register.ACCEL_CONFIG) >> 3);
+                if (_accelerometerRange != value)
+                    throw new IOException($"Can set {nameof(AccelerometerRange)}, desired value {value}, stored value {_accelerometerRange}");
+            }
+        }
+
+        /// <summary>
+        /// Get or set the accelerometer bandwidth
+        /// </summary>
+        public AccelerometerBandwidth AccelerometerBandwidth
+        {
+            get => _accelerometerBandwidth;
+
+            set
+            {
+                WriteRegister(Register.ACCEL_CONFIG_2, (byte)value);
+                _accelerometerBandwidth = (AccelerometerBandwidth)ReadByte(Register.ACCEL_CONFIG_2);
+                if (_accelerometerBandwidth != value)
+                    throw new IOException($"Can set {nameof(AccelerometerBandwidth)}, desired value {value}, stored value {_accelerometerBandwidth}");
+            }
+        }
+
+        /// <summary>
+        /// Get the real accelerometer bandwith. This allows to calculate the real
+        /// degree per second
+        /// </summary>
+        public float AcceleromterConversion
+        {
+            get
+            {
+                float val = 0;
+                switch (AccelerometerRange)
+                {
+                    case AccelerometerRange.Range02G:
+                        val = 2.0f;
+                        break;
+                    case AccelerometerRange.Range04G:
+                        val = 4.0f;
+                        break;
+                    case AccelerometerRange.Range08G:
+                        val = 8.0f;
+                        break;
+                    case AccelerometerRange.range16G:
+                        val = 16.0f;
+                        break;
+                    default:
+                        break;
+                }
+
+                val = (val * Gravity) / Adc;
+                return val / (1 + SampleRateDivider);
+            }
+        }
+
+        /// <summary>
+        /// Get the acccelerometer in G
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///         +X
+        /// </summary>
+        public Vector3 Accelerometer => GetRawAccelerometer() * AcceleromterConversion;
```

Comment:
This shouldn't be a property, and instead it should be a GetAccelerometer method. We had some discussion about this here: [https://github.com/dotnet/iot/issues/626](https://github.com/dotnet/iot/issues/626)

Reply:
Same comment for the rest of the readings. I'm fine with keeping properties for the settings, like AccelerometerLowPowerFrequency, but all the ones that are actually reading sensor data should really be properties instead.

Reply:
Yes, I will adjust all of them